[Bug]: Crash in CleanupNetwork with nil value

### Testcontainers version

v0.34.0

### Using the latest Testcontainers version?

Yes

### Host OS

Linux

### Host arch

amd64

### Go version

1.23.4

### Docker version

```shell
Client:
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.23.1
 Git commit:        ce1223035a
 Built:             Wed Sep 25 14:43:43 2024
 OS/Arch:           linux/amd64
 Context:           default

Server:
 Engine:
  Version:          27.3.1
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.23.1
  Git commit:       41ca978a0a
  Built:            Wed Sep 25 14:43:43 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v2.0.1
  GitCommit:        88aa2f531d6c2922003cc7929e51daf1c14caa0a.m
 runc:
  Version:          1.2.3
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### Docker info

```shell
Client:
 Version:    27.3.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  0.19.2
    Path:     /usr/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  2.32.1
    Path:     /usr/lib/docker/cli-plugins/docker-compose

Server:
 Containers: 2
  Running: 1
  Paused: 0
  Stopped: 1
 Images: 113
 Server Version: 27.3.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 88aa2f531d6c2922003cc7929e51daf1c14caa0a.m
 runc version: 
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.12.4-arch1-1
 Operating System: Arch Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 24
 Total Memory: 62.68GiB
 Name: p200300EEEF.fab
 ID: 68738235-b89a-415a-be22-29f7ba78cb7e
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: l0nax
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

### What happened?

When calling `testcontainers.CleanupNetwork` with a `nil` value for `network`, the test panics.
This is despite the godoc of the function stating:
> [...] If network is nil, its a no-op.

Example test:
```go
package main

import (
	"testing"

	"github.com/testcontainers/testcontainers-go"
)

func TestBug(t *testing.T) {
	var network *testcontainers.DockerNetwork
	testcontainers.CleanupNetwork(t, network)
}
```

### Relevant log output

```shell
=== RUN   TestBug
--- FAIL: TestBug (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x6d9f93]

goroutine 40 [running]:
testing.tRunner.func1.2({0x72be40, 0xa26390})
        /home/eb/.gvm/gos/go1.23.3/src/testing/testing.go:1632 +0x230
testing.tRunner.func1()
        /home/eb/.gvm/gos/go1.23.3/src/testing/testing.go:1635 +0x35e
panic({0x72be40?, 0xa26390?})
        /home/eb/.gvm/gos/go1.23.3/src/runtime/panic.go:785 +0x132
github.com/testcontainers/testcontainers-go.(*DockerNetwork).Remove(0x0, {0x82a600, 0xa58560})
        /home/eb/.go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.34.0/docker.go:889 +0x33
github.com/testcontainers/testcontainers-go.CleanupNetwork.func1()
        /home/eb/.go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.34.0/testing.go:89 +0x3c
testing.(*common).Cleanup.func1()
        /home/eb/.gvm/gos/go1.23.3/src/testing/testing.go:1176 +0x109
testing.(*common).runCleanup(0xc000168680, 0x0?)
        /home/eb/.gvm/gos/go1.23.3/src/testing/testing.go:1354 +0xd5
testing.tRunner.func2()
        /home/eb/.gvm/gos/go1.23.3/src/testing/testing.go:1684 +0x25
testing.tRunner(0xc000168680, 0x7ce068)
        /home/eb/.gvm/gos/go1.23.3/src/testing/testing.go:1696 +0x11e
created by testing.(*T).Run in goroutine 1
        /home/eb/.gvm/gos/go1.23.3/src/testing/testing.go:1743 +0x390
FAIL    gitlab.com/l0nax/go-dump/testcontainers-bug-report-nil-network  0.194s
FAIL
```

### Additional information

_No response_

--- a/network/network_test.go
+++ b/network/network_test.go
@@ -440,3 +440,8 @@ func TestWithNewNetworkContextTimeout(t *testing.T) {
 	require.Empty(t, req.Networks)
 	require.Empty(t, req.NetworkAliases)
 }
+
+func TestCleanupWithNil(t *testing.T) {
+	var network *testcontainers.DockerNetwork
+	testcontainers.CleanupNetwork(t, network)
+}

--- a/testing.go
+++ b/testing.go
@@ -83,7 +83,9 @@ func CleanupNetwork(tb testing.TB, network Network) {
 	tb.Helper()
 
 	tb.Cleanup(func() {
-		noErrorOrIgnored(tb, network.Remove(context.Background()))
+		if !isNil(network) {
+			noErrorOrIgnored(tb, network.Remove(context.Background()))
+		}
 	})
 }
 

